extensions/amp-analytics/0.1/transport.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {log} from '../../../src/log';
+import {loadPromise} from '../../../src/event-helper';
+
+/** @const {string} */
+const TAG_ = 'AmpAnalytics.Transport';
+
+/**
+ * @param {!Window} win
+ * @param {string} request
+ * @param {!Array<string>} transports
+ */
+export function sendRequest(win, request, transports) {
+  for (let i = 0; i < transports.length; ++i) {
+    if (transports[i] == 'image') {
+      return sendRequestUsingImage(win, request);
+    } else if (transports[i] == 'beacon') {
+      if (sendRequestUsingBeacon(win, request)) {
+        return;
+      }
+    } else if (transports[i] == 'xhrpost') {
+      if (sendRequestUsingXhr(win, request)) {
+        return;
+      }
+    } else {
+      log.warn(TAG_, 'Invalid transport', transports[i]);
+    }
+  }
+  log.warn(TAG_, 'Failed to send request', request);
+}
+
+/**
+ * @param {!Window} win
+ * @param {string} request
+ */
+function sendRequestUsingImage(win, request) {
+  const image = new Image();
+  image.src = request;
+  image.width = 1;
+  image.height = 1;
+  loadPromise(image).then(() => {
+    log.fine(TAG_, 'Sent image request', request);
+  }).catch(() => {
+    log.warn(TAG_, 'Failed to send image request', request);
+  });
+}
+
+/**
+ * @param {!Window} win
+ * @param {string} request
+ * @return {boolean} True if this browser supports navigator.sendBeacon.
+ */
+function sendRequestUsingBeacon(win, request) {
+  if (!win.navigator.sendBeacon) {
+    return false;
+  }
+  win.navigator.sendBeacon(request, '');
```

Comment:
I believe sendBeacon returns true/false based on whether it succeeds. Should we use it to continue the fallback?